src/filter_iterator.c

```diff
@@ -458,15 +458,15 @@ static void twa_fillEmptyBuckets(size_t *write_index,
         } else if (n_samples_after > 1) {
             timestamp_t delta = sample_afAfter.timestamp - sample_after.timestamp;
             if (tb + (delta / 2) <= sample_after.timestamp) {
-                aggregation->finalizeEmpty(&val);
+                aggregation->finalizeEmpty(NULL, &val);
```

Comment:
we need tests to cover this

Reply:
We don’t miss tests, this is a dead code, I didn’t want to remove since we might want it in the future.
It wasn’t dead before Lior
 asked me to change the TWA behaviour